src/components/global-view-core.js

```diff
@@ -269,6 +275,20 @@ const GlobalViewCore = ({ timestampMin, timestampMax, timestampStep }) => {
         getMissingData,
     ]);
 
+    useEffect(() => {
+        let index = openEvent.indexOf(true);
+        if (index >= 0) {
+            fetchTimestampData(
+                new Date(steps[index].timestamp).toISOString(),
+                intlRef,
+                enqueueSnackbar
+            ).then((res) => {
+                steps[index].taskData.processEvents = res.processEvents;
+                setIsLoadingEvent(false);
+            });
+        }
+    }, [isLoadingEvent, enqueueSnackbar, intlRef, openEvent, steps]);
```

Comment:
I don't understand why you use a useEffect to launch the request. Here we want to launch the request when we click on the event button, so the callback corresponding to the button's onClick property is called "handleEventOpen", you just have to do the "fetch" in this callback.

In addition, the dependency list of your useEffect is not good and caused another problem, the request being launched twice instead of once.

Before the click

![Capture d’écran de 2023-10-11 11-26-30](https://github.com/farao-community/gridcapa-app/assets/116628858/5c805967-4c04-4011-a405-1cd8b18606a9)

After the click

![Capture d’écran de 2023-10-11 11-26-43](https://github.com/farao-community/gridcapa-app/assets/116628858/4f758d4b-8e75-4c55-b974-ac306dbcaba9)

We can notice that the query was also launched three times when arriving on the page...
But this is not due to your development.